src/common/services/Database/subscribers/TemperatureBreachSubscriber.ts

```diff
@@ -0,0 +1,77 @@
+/* istanbul ignore file */
+
+/* eslint-disable import/no-cycle */
+import {
+  EventSubscriber,
+  EntitySubscriberInterface,
+  AfterInsert,
+  AfterUpdate,
+  InsertEvent,
+  UpdateEvent,
+} from 'typeorm/browser';
+
+import { TemperatureBreach, TemperatureBreachConfiguration } from '../entities';
+
+import { SyncService } from '../../SyncService';
+import { ENTITIES } from '../../../constants';
+
+@EventSubscriber()
+class TemperatureBreachSubscriber implements EntitySubscriberInterface {
+  public listenTo() {
+    return TemperatureBreach;
+  }
+
+  public async updateSyncQueue(event: InsertEvent<any> | UpdateEvent<any>) {
+    const { entity, manager, metadata } = event;
+
+    const {
+      id,
+      startTimestamp,
```

Comment:
Note no `endTimestamp`, the mSupply schema doesn't like when this is passed as a `null`. Got frustrated by tyhis so left it for now. 

Options here are to add logic to only add `endTimestamp` if it is non-null, or to update mSupply schema to be a bit more flexible (e.g. strip out `null` fields before validating against schema).

Reply:
couple things for this one (we can decide what actually needs to be done for this iteration)

- We should just add the `threshold`/`type` fields to the schema
- `type` field to config
- I think adding in the `endTimestamp` here is OK but also in mSupply is fine. Up to whatever you can be bothered doing